src/Distortion.cpp

```diff
@@ -233,5 +259,69 @@ void applyDistortion(double ux, double uy, double &dx, double &dy,
       computeTransverseDistortion(ux, uy, dx, dy, opticalDistCoeffs);
     }
     break;
+    case KAGUYATC: {
+      if (opticalDistCoeffs.size() != 8) {
+        throw "Distortion coefficients for Kaguya TC must be of size 8, got: " +  std::to_string(opticalDistCoeffs.size());
+      }
+
+      const double* odkx = opticalDistCoeffs.data();
+      const double* odky = opticalDistCoeffs.data()+4;
+
+      double xt = ux;
+      double yt = uy;
+
+      double xx, yy, r, rr, rrr, dr_x, dr_y;
+      double xdistortion, ydistortion;
+      double xdistorted, ydistorted;
+      double xprevious, yprevious;
+
+      xprevious = 1000000.0;
+      yprevious = 1000000.0;
+
+      double tolerance = 0.000001;
+      bool bConverged = false;
+
+      // Iterating to introduce distortion...
+      // We stop when the difference between distorted coordinates
+      // in successive iterations is below the given tolerance
+      for (int i = 0; i < 50; i++) {
+        xx = xt * xt;
+        yy = yt * yt;
+        rr = xx + yy;
+        r = sqrt(rr);
+        rrr = rr * r;
+
+        // Radial distortion
+        // dr is the radial distortion contribution
+        dr_x = odkx[0] + odkx[1] * r + odkx[2] * rr + odkx[3] * rrr; // why did hayabusa have a -1
```

Comment:
I also forgot to take the comment about "why did hayabusa have a -1" out of the ISIS code.